modules/core/src/copy.cpp

```diff
@@ -826,6 +826,50 @@ void flip( InputArray _src, OutputArray _dst, int flip_mode )
         flipHoriz( dst.ptr(), dst.step, dst.ptr(), dst.step, dst.size(), esz );
 }
 
+#ifdef HAVE_OPENCL
+
+static bool ocl_rotate(InputArray _src, OutputArray _dst, int rotateMode)
+{
+    switch (rotateMode)
+    {
+    case ROTATE_90:
+        flip(_src.getUMat().t(), _dst, 1);
```

Comment:
this implementation is inefficient, it allocates a temporary array for transposed image. More efficiently would be to write it as:

<pre>
transpose(_src, _dst);
flip(_dst, _dst, 1);
</pre>